cpp/meson.options

```diff
@@ -29,6 +36,20 @@ option(
     value: false,
 )
 
+option(
+    'filesystem',
+    type: 'boolean',
+    description: 'Build the Arrow Filesystem Layer',
+    value: false,
+)
+
+option(
+    'gcs',
+    type: 'boolean',
+    description: 'Build Arrow with GCS support (requires the GCloud SDK for C++)',
```

Comment:
Based on its README: https://github.com/googleapis/google-cloud-cpp?tab=readme-ov-file#google-cloud-platform-c-client-libraries

```suggestion
    description: 'Build Arrow with GCS support (requires the Google Cloud Platform C++ Client Libraries)',
```

Reply:
No problem - do you want me to update `DefineOptions.cmake` with the new description as well?

Reply:
Yes, please.